olad/RDMHTTPModule.cpp

```diff
@@ -3313,6 +3322,164 @@ string RDMHTTPModule::SetDnsDomainName(const HTTPRequest *request,
   return error;
 }
 
+/**
+ * @brief Handle the request for the dimmer curve section.
+ */
+string RDMHTTPModule::GetCurve(OLA_UNUSED const HTTPRequest *request,
+                                      HTTPResponse *response,
+                                      unsigned int universe_id,
+                                      const UID &uid,
+                                      bool include_descriptions) {
+  string error;
+
+  curve_info *info = new curve_info;
+  info->universe_id = universe_id;
+  info->uid = new UID(uid);
+  info->include_descriptions = include_descriptions;
+  info->active = 0;
+  info->next = 1;
+  info->total = 0;
+
+  m_rdm_api.GetCurve(
+      universe_id,
+      uid,
+      ola::rdm::ROOT_RDM_DEVICE,
+      NewSingleCallback(this,
+                        &RDMHTTPModule::GetCurveHandler,
+                        response,
+                        info),
+      &error);
+  return error;
+}
+
+/**
+ * @brief Handle the response to a dimmer curve call and build the curve info
+ * object
+ */
+void RDMHTTPModule::GetCurveHandler(
+    HTTPResponse *response,
+    curve_info *info,
+    const ola::rdm::ResponseStatus &status,
+    uint8_t active_curve,
+    uint8_t curve_count) {
+  if (CheckForRDMError(response, status)) {
+    delete info->uid;
+    delete info;
+    return;
+  }
+
+  info->active = active_curve;
+  info->total = curve_count;
+
+  if (info->include_descriptions) {
+    GetNextCurveDescription(response, info);
+  } else {
+    SendCurveResponse(response, info);
+  }
+}
+
+/**
+ * @brief Handle the response to a curve description call and add to curve
+ * info object
+ */
+void RDMHTTPModule::GetNextCurveDescription(HTTPResponse *response,
+                                            curve_info *info) {
+  string error;
+  while (info->next <= info->total) {
+    bool r = m_rdm_api.GetCurveDescription(
+        info->universe_id,
+        *(info->uid),
+        ola::rdm::ROOT_RDM_DEVICE,
+        info->next,
+        NewSingleCallback(this,
+                          &RDMHTTPModule::GetCurveDescriptionHandler,
+                          response,
+                          info),
+        &error);
+    if (r) {
+      return;
+    }
+
+    info->next++;
+  }
+
+  SendCurveResponse(response, info);
+}
+
+void RDMHTTPModule::GetCurveDescriptionHandler(
+    HTTPResponse *response,
+    curve_info *info,
+    const ola::rdm::ResponseStatus &status,
+    OLA_UNUSED uint8_t curve,
+    const string &resp_description) {
+  string description = "";
+
+  if (CheckForRDMSuccess(status)) {
+    description = resp_description;
+  }
+
+  info->curves.push_back(pair<uint32_t, string>(curve, description));
+
+  if (info->next == info->total) {
+    SendCurveResponse(response, info);
+  } else {
+    info->next++;
+    GetNextCurveDescription(response, info);
+  }
+}
+
+/**
+ * @brief Build the curve http response
+ */
+void RDMHTTPModule::SendCurveResponse(HTTPResponse *response,
```

Comment:
I'll try to find a way to harden this up

Reply:
Thanks. It's probably as simple as just storing an empty string for the description if the responder doesn't respond or NAcks. But you therefore probably also ought to handle a timeout on some curve descriptions. The Dummy Responders are a good test ground for this, as you can hack them around to do odd things fairly easily.

Reply:
Testing with the dummy responder, this is just storing an empty string and not crashing or anything. Works both with an empty curve_description response and just never getting a response, so I think this should be fine.

Reply:
Added a solution where it will simply return the index of the curve if the device doesn't return a description or returns a 0 length name. I think that should address your concerns?